### PR TITLE
add codeql to ci

### PR DESCRIPTION
**Pull request by two-heart**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: "CodeQL PR Analysis"
+
+on:
+    workflow_call:
+    workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze (${{matrix.language}})
+    runs-on: 'ubuntu-latest'
```
Comment (ripatel-fd) [inline .github/workflows/codeql.yml:10]: If this check takes longer than 2 minutes or so, we should run it on-prem

Comment (two-heart) [reply .github/workflows/codeql.yml:10]: I agree, see slack channel. I just can't test on on-prem runners and don't want to PR something I am not sure is working.

**Review by topointon-jump — DISMISSED** [review]
Following Anway's suggestion, we will run this on GitHub runners for now. Then when this is stable we will move it to on-prem.

**Review by anwayde — DISMISSED** [review]

**Comment by ravyu-jump** [comment]
@two-heart looks like the workflow script is broken?

**Comment by two-heart** [comment]
Fixed the workflow file. Let me tune some queries such that we are maybe a bit faster.

**Review by ravyu-jump — APPROVED** [review]

**Pull request merged**